janitor/src/main/scala/com/broadinstitute/dsp/janitor/NodepoolRemover.scala

```diff
@@ -24,17 +24,15 @@ object NodepoolRemover {
       ): F[Option[Nodepool]] =
         for {
           ctx <- ev.ask
-          _ <-
-            if (!isDryRun) {
-              n.cloudContext match {
-                case CloudContext.Gcp(value) =>
-                  val msg = DeleteNodepoolMeesage(n.nodepoolId, value, Some(ctx))
-                  deps.publisher.publishOne(msg)
-                case CloudContext.Azure(_) =>
-                  logger.warn("Azure is not supported yet") // TODO: IA-3623
-              }
-
-            } else F.unit
-        } yield Some(n)
+          res <-
```

Comment:
Changed this logic (kind of inverted the `isDryRun` and `cloudContext` check) to be consistent with the others. Now we only return `Some` for the GCP case; and only publish if `isDryRun` is true.